banyand/liaison/http/server.go

```diff
@@ -130,7 +134,7 @@ func (p *service) Serve() run.StopNotify {
 	p.mux.Mount("/api", http.StripPrefix("/api", gwMux))
 	go func() {
 		p.l.Info().Str("listenAddr", p.listenAddr).Msg("Start liaison http server")
-		if err := p.srv.ListenAndServe(); err != http.ErrServerClosed {
+		if err := http.ListenAndServeTLS(p.listenAddr, p.certFile, p.keyFile, p.srv.Handler); err != http.ErrServerClosed {
```

Comment:
You had better add conditional branches to boot a TLS server or plain server based on `p.tls`.

Reply:
use `p.svr.ListenAndServerTLS` in place of `http.ListenAndServerTLS`